rs_server/s3_storage_handler/s3_storage_handler.py

```diff
@@ -17,243 +24,719 @@
 S3_ERR_FORBIDDEN_ACCESS = 403
 S3_ERR_NOT_FOUND = 404
 
-s3_client_mutex = Lock()
 
+class S3StorageHandler:
+    """Interact with an S3 storage
+
+    S3StorageHandler for interacting with an S3 storage service.
+
+    Attributes:
+        access_key_id (str): The access key ID for S3 authentication.
+        secret_access_key (str): The secret access key for S3 authentication.
+        endpoint_url (str): The endpoint URL for the S3 service.
+        region_name (str): The region name.
+
+    Methods:
+        __init__(self, access_key_id, secret_access_key, endpoint_url, region_name):
+            Initializes the S3StorageHandler instance, establishes a connection to S3, and sets up logging.
+
+        __get_s3_client(self, access_key_id, secret_access_key, endpoint_url, region_name):
+            Private function to retrieve or create the S3 client instance.
+
+        connect_s3(self):
+            Establishes a connection to the S3 service.
+
+        disconnect_s3(self):
+            Closes the connection to the S3 service.
+
+        delete_file_from_s3(self, bucket, s3_obj):
+            Deletes a file from S3.
+
+        get_secrets(secrets, secret_file, logger=None):
+            Reads secrets from a specified file.
+
+        get_basename(input_path):
+            Gets the filename from a full path.
+
+        files_to_be_downloaded(self, bucket, paths):
+            Creates a list of S3 keys to be downloaded.
+
+        files_to_be_uploaded(self, paths):
+            Creates a list of local files to be uploaded.
+
+        list_s3_files_obj(self, bucket, prefix):
+            Retrieves the content of an S3 directory.
+
+        get_s3_data(s3_url):
+            Parses S3 URL to extract bucket, prefix, and file.
+
+        check_bucket_access(self, bucket):
+            Checks the accessibility of an S3 bucket.
```

Comment:
This "Method" section is useless (because the methods docstrings are available below) and will be too difficult to maintain (i.e. when someone will implement a new method, they will forget to add it in the "Method" section). I suggest we remove it.

Reply:
Removed it